app/pods/publications/publication/documents/template.hbs

```diff
@@ -89,13 +88,13 @@
                     {{#if (gt (await piece.agendaitems.length) 0)}}
                       <WebComponents::AuIcon @name="lock-closed" @iconSkinColor="muted" @layout="icon-only"/>
                     {{/if}}
-                    {{await piece.name}}
+                    {{piece.name}}
                   </td>
                   <td colspan="2">
                     {{await piece.file.extension}}
                   </td>
                   <td colspan="2">
-                    {{await piece.documentContainer.type.label}}
+                    {{piece.documentContainer.type.label}}
```

Comment:
Hier ben ik niet zeker van. Ik merk dat, ondanks de "include"-parameter de get('documentContainer.type') niet steeds werkt zonder await in de controller. Ik begrijp niet goed waarom. Ik weet niet of dit in templates ook zo is.

Reply:
Zou moeten werken zonder await, de includes in de route.js hebben piece.documentContainer.type al geladen

Reply:
Dit is prima zo. Merk op dat ember-data relaties geen pure promise, maar een [PromiseProxy-achtig](https://api.emberjs.com/ember/release/classes/PromiseProxyMixin) object returnen. Wanneer je weet dat:
- promiseproxies `null` returnen zolang ze unresolved zijn en de resolved waarde returnen eens ze resolved zijn
- ember templates falsy values (`null`, ...) niet renderen

merk je dus dat indien deze elementen niet gerendered zullen worden als de relatie nog aan het laden is, maar automatisch zullen tevoorschijn komen eens geladen.  
Voor een gewone weergave is het dus steeds ok om geen `await` te gebruiken. Dit is wel anders wanneer je deze waarde zou willen doorgeven aan een component. Daar hangt het er van af of een component om kan met promises (die binnen de component nog resolved moeten worden). Wij kiezen er gewoonlijk voor om componenten zo te definieren dat ze enkel resolvede waarden (geen promises) als argumenten aanvaarden.